iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/relative_pointer.inl

```diff
@@ -67,13 +67,15 @@ inline T* RelativePointer<T>::operator->() const noexcept
 template <typename T>
 inline T* RelativePointer<T>::get() const noexcept
 {
-    return static_cast<T*>(computeRawPtr());
+    auto ptr = static_cast<T*>(computeRawPtr());
+    cxx::Ensures(ptr != nullptr);
```

Comment:
This runtime check is bad IMO. It is also not like smart pointers work, e.g. STL `shared_ptr` and friends. I am against this check, not having this check is a conscious design decision.

Reply:
See my [answer](https://github.com/eclipse-iceoryx/iceoryx/pull/1597#discussion_r959976184) above. I think there was a misunderstanding. Closing here, so we can continue the discussion in the other thread.